tests/models/wav2vec2_conformer/test_modeling_wav2vec2_conformer.py

```diff
@@ -922,65 +1020,88 @@ def test_inference_ctc_normal_batched_rope(self):
+        # fmt: off
+        expected_slice_0 = torch.tensor(
+            [[-0.0098, -0.0570, -0.1286,  0.0439, -0.1037, -0.0235],
+            [-0.0767,  0.0574, -0.3224,  0.0482,  0.0440, -0.0193],
+            [ 0.0220, -0.0878, -0.2027, -0.0028, -0.0666,  0.0721],
+            [ 0.0307, -0.1099,  0.0273, -0.0416, -0.0715,  0.0094],
+            [ 0.0758, -0.0291,  0.1084,  0.0004, -0.0751, -0.0116],
+            [ 0.0349, -0.0343, -0.0098,  0.0415, -0.0617,  0.0241],
+            [-0.0193, -0.0171,  0.1965,  0.0797, -0.0308,  0.2033],
+            [-0.0323, -0.0315,  0.0948,  0.0944, -0.0254,  0.1241],
+            [-0.0493,  0.0010, -0.1762,  0.0034, -0.0787,  0.0832],
+            [ 0.0043, -0.1228, -0.0739,  0.0266, -0.0337, -0.0068]]
+        ).to(torch_device)
+        # fmt: on
+
+        # fmt: off
+        expected_slice_1 = torch.tensor(
+            [[-0.0348, -0.0521, -0.3036,  0.0285, -0.0715, -0.0453],
+            [-0.0102,  0.0114, -0.3266,  0.0027, -0.0558,  0.0038],
+            [ 0.0454,  0.0148, -0.2418, -0.0392, -0.0455,  0.0478],
+            [-0.0013,  0.0825, -0.1730, -0.0091, -0.0426,  0.0360],
+            [-0.0227,  0.0687, -0.1168,  0.0569, -0.0160,  0.0759],
+            [-0.0318,  0.0562, -0.0508,  0.0605,  0.0150,  0.0953],
+            [-0.0415,  0.0438,  0.0233,  0.0336,  0.0262,  0.0860],
+            [-0.0163,  0.0048,  0.0807,  0.0119,  0.0712,  0.0158],
+            [ 0.0244, -0.0145,  0.0262, -0.0237,  0.0283, -0.0125],
+            [-0.0587, -0.0516, -0.0368, -0.0196,  0.0307, -0.1434]]
+        ).to(torch_device)
+        # fmt: on
+
+        self.assertTrue((outputs.last_hidden_state[0, 25:35, 4:10] - expected_slice_0).abs().max() <= 1e-4)
+        self.assertTrue((outputs.last_hidden_state[1, 25:35, 4:10] - expected_slice_1).abs().max() <= 1e-4)
+
+        self.assertAlmostEqual(outputs.last_hidden_state[1].mean().item(), 3.3123e-05)
+        self.assertAlmostEqual(outputs.last_hidden_state[1].std().item(), 0.1545, delta=2e-5)
+
+        self.assertListEqual(list(outputs.last_hidden_state.shape), [2, 326, 1024])
+
+    def test_inference_w2v2_bert_pretrained(self):
+        model_id = "facebook/w2v-bert-2.0"
+        model = Wav2Vec2ConformerForPreTraining.from_pretrained(model_id)
+        model.to(torch_device)
+        feature_extractor = AutoFeatureExtractor.from_pretrained(model_id)
+        input_name = feature_extractor.model_input_names[0]
 
-        # retrieve cosine sim of masked features
-        cosine_sim_masked = cosine_sim[mask_time_indices]
+        input_speech = self._load_datasamples(2)
+        inputs = feature_extractor(input_speech, return_tensors="pt", padding=True).to(torch_device)
+        features_shape = inputs["input_features"].shape[:2]
 
-        # ... now compare to randomly initialized model
+        # apply augmentation
+        model.wav2vec2_conformer.config.apply_spec_augment = True
```

Comment:
Is this deterministic?

Reply:
It's not, but `check_inference_pretrained` sets the seed !